// @material-tailwind/react/dist/theme/badge.cjs
"use strict";
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// src/theme/badge.ts
var badge_exports = {};
__export(badge_exports, {
  badgeContentTheme: () => badgeContentTheme,
  badgeIndicatorTheme: () => badgeIndicatorTheme,
  badgeTheme: () => badgeTheme
});
module.exports = __toCommonJS(badge_exports);
var badgeTheme = {
  defaultProps: {
    color: "primary",
    overlap: "square",
    placement: "top-end"
  },
  baseStyle: "relative inline-flex"
};
var badgeContentTheme = {
  baseStyle: ""
};
var badgeIndicatorTheme = {
  baseStyle: `
    absolute px-1 py-0.5 text-xs border leading-none grid place-items-center rounded-full min-w-[10px] min-h-[10px]

    data-[placement='top-start']:data-[overlap='square']:top-[6%] data-[placement='top-start']:data-[overlap='square']:left-[6%]
    data-[placement='top-start']:data-[overlap='square']:-translate-x-1/2 data-[placement='top-start']:data-[overlap='square']:-translate-y-1/2
    data-[placement='top-start']:data-[overlap='circular']:top-[14%] data-[placement='top-start']:data-[overlap='circular']:left-[14%]
    data-[placement='top-start']:data-[overlap='circular']:-translate-x-1/2 data-[placement='top-start']:data-[overlap='circular']:-translate-y-1/2
    
    data-[placement='top-end']:data-[overlap='square']:top-[6%] data-[placement='top-end']:data-[overlap='square']:right-[6%]
    data-[placement='top-end']:data-[overlap='square']:translate-x-1/2 data-[placement='top-end']:data-[overlap='square']:-translate-y-1/2
    data-[placement='top-end']:data-[overlap='circular']:top-[14%] data-[placement='top-end']:data-[overlap='circular']:right-[14%]
    data-[placement='top-end']:data-[overlap='circular']:translate-x-1/2 data-[placement='top-end']:data-[overlap='circular']:-translate-y-1/2
    
    data-[placement='bottom-start']:data-[overlap='square']:bottom-[6%] data-[placement='bottom-start']:data-[overlap='square']:left-[6%]
    data-[placement='bottom-start']:data-[overlap='square']:-translate-x-1/2 data-[placement='bottom-start']:data-[overlap='square']:translate-y-1/2
    data-[placement='bottom-start']:data-[overlap='circular']:bottom-[14%] data-[placement='bottom-start']:data-[overlap='circular']:left-[14%]
    data-[placement='bottom-start']:data-[overlap='circular']:-translate-x-1/2 data-[placement='bottom-start']:data-[overlap='circular']:translate-y-1/2
    
    data-[placement='bottom-end']:data-[overlap='square']:bottom-[6%] data-[placement='bottom-end']:data-[overlap='square']:right-[6%]
    data-[placement='bottom-end']:data-[overlap='square']:translate-x-1/2 data-[placement='bottom-end']:data-[overlap='square']:translate-y-1/2
    data-[placement='bottom-end']:data-[overlap='circular']:bottom-[14%] data-[placement='bottom-end']:data-[overlap='circular']:right-[14%]
    data-[placement='bottom-end']:data-[overlap='circular']:translate-x-1/2 data-[placement='bottom-end']:data-[overlap='circular']:translate-y-1/2
  `,
  color: {
    primary: "bg-primary border-primary text-primary-foreground",
    secondary: "bg-secondary border-secondary text-secondary-foreground",
    info: "bg-info border-info text-info-foreground",
    success: "bg-success border-success text-success-foreground",
    warning: "bg-warning border-warning text-warning-foreground",
    error: "bg-error border-error text-error-foreground"
  }
};
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  badgeContentTheme,
  badgeIndicatorTheme,
  badgeTheme
});
